tokenclient/tokenFlows.go

```diff
@@ -128,14 +145,48 @@ func (t *TokenFlows) getURL(customerTenantURL string) (string, error) {
 	return "", fmt.Errorf("customer tenant url '%v' can't be parsed: %w", customerTenantURL, err)
 }
 
-func (t *TokenFlows) performRequest(r *http.Request, v interface{}) error {
-	res, err := t.Options.HTTPClient.Do(r)
+func (t *TokenFlows) getOrRequestToken(r request) (string, error) {
+	// token cached?
+	cachedToken := t.readFromCache(&r)
+	if cachedToken != "" {
+		return cachedToken, nil
+	}
+
+	// request token
+	var tokenRes tokenResponse
+	err := t.performRequest(r, &tokenRes)
+	if err != nil {
+		return "", err
+	}
+	if tokenRes.Token == "" {
+		return "", fmt.Errorf("error parsing requested client credentials token: no 'access_token' property provided")
+	}
+
+	// cache and return retrieved token
+	t.writeToCache(r, tokenRes.Token)
+	return t.readFromCache(&r), nil
```

Comment:
Why not just returning tokenRes.Token?

Reply:
maybe ....I wanted to test implicitly that the token was sucessfully written to cache... but this isn't performant as just returning the token.